### DISCUSSION TITLE
Questions on Probability Scaling and Model Constraints in biomod2

### DISCUSSION THREAD

**Discussion by BenWalterGit** [biomodhub/biomod2#492] [General]
Hi biomod2 team,

I’ve been using biomod2 for a while now, and I find the package fantastic. However, I’ve recently encountered a couple of questions and would appreciate your insights.

Scaling of Probabilities: From my understanding, the probability output from the models represent the likelihood of species occurrence in each pixel of the study area. However, in my current project, the probabilities produced by an ensemble model are confined to a very narrow range (e.g., between 0.4 and 0.6). Is there a function within biomod2 that can scale these probabilities to cover the full range from 0 to 1? I am specifically referring to scaling during model fitting and not projection, as I’m particularly interested in the response curves.

Bounding Function for Model Fitting: Is it possible to constrain the model fitting process within biomod2 using specific boundary values or functions? For instance, if I lack data to model the entire niche of a plant species but know that it cannot survive temperatures below -5°C, is there a way to set such a boundary condition so that the model converges towards this limit? If so, for which algorithms would this exist?

Thank you in advance for your feedback—it’s greatly appreciated.

Best regards,
Ben

**Comment by HeleneBlt** [comment]
Hello Ben ! 

Thanks a lot for your interest 😊 

**Scaling of Probabilities:** No, there isn't some function to rescale. However, this means you did an ensemble model with either : 

- models in serious disagreement 
- models already within this narrow range 

You can have a look at your single models, look where they diverge, and where they are similar and perhaps make a new selection of single models for your ensemble models.
Using the committee averaging ensemble models can also be useful. 

**Bounding Function for Model Fitting:** There isn't a special argument for that. However, you can choose special absences or pseudo-absences to be sure to integrate the temperature range and help the models to define some limits. The SRE pseudo absences method will be useful here or you can create your own PAtable (see [bm_PseudoAbsences](https://biomodhub.github.io/biomod2/reference/bm_PseudoAbsences.html) ).
You can also create a new variable. For example, `temperature_range` with a boolean value if the localisation is in the good range or not. In this case, you need algorithms which accept categorical variables so you need to remove SRE or XGBOOST. 

Don't hesitate if it is not clear. 
Hélène

**Comment by BenWalterGit** [comment]
Hi Hélène

Sorry for the late reply. But your answers helped me a lot. Especially the piece on model disagreement, I didn't think of that at all.

Thank you.

Best Benjamin
